cabal-install/Distribution/Client/SetupWrapper.hs

```diff
@@ -338,11 +342,28 @@ runSetupMethod SelfExecMethod = selfExecSetupMethod
 runSetup :: Verbosity -> Setup
          -> [String]  -- ^ command-line arguments
          -> IO ()
-runSetup verbosity setup args =
-  let method = setupMethod setup
-      options = setupScriptOptions setup
-      bt = setupBuildType setup
-  in runSetupMethod method verbosity options bt args
+runSetup verbosity setup args = do
+    runSetupMethod method verbosity options bt args'
+  where
+    method = setupMethod setup
+    options = setupScriptOptions setup
+    bt = setupBuildType setup
+
+    -- Hack to rewrite new-style --verbose flags
+    -- if they happen to reach this point for pre-Cabal-1.25
+    --
+    -- TODO: figure out a less hacky-way to accomplish this without
```

Comment:
For the record, I ended up committing this very functionality, with a similar hack, in eb0593ed7690e402c288011dc77964e404bb78d9

The method @dcoutts suggests doesn't actually work, because there is a Verbosity flag per command you might run. So you'd have to make sure you applied it to every *Flags data structure, not just ConfigFlags. So there's a phase ordering problem, where you don't know the version of your Setup script until you're in SetupWrapper, but at that point it's too late to fix Verbosity.